### DISCUSSION TITLE
Gopro Wide Extreme saturation after convertion

### DISCUSSION THREAD

**Discussion by markusfinholt** [xtremestuff/protune-transforms#11] [Q&A]
### I tried out Gopro Wide+LOGB=400 to DWG and it gave me some really pink and saturated results.

Here is the normal shot without wide enabled. Only LOGB=400
![WIDE=0_1 110 3](https://github.com/xtremestuff/protune-transforms/assets/76013032/c2f9ffb9-972f-43a0-987d-f9e5dd6c4e42)

Looks fine, colors are accurate after adding contrast.

Here is WIDE=1 added. But no convertion.
![Wide_Raw](https://github.com/xtremestuff/protune-transforms/assets/76013032/90132b7f-ea99-4fd1-99f8-e8f7ca170225)

Here is the WIDE=1 enabled, converted to DWG, and output to REC709, Gamma 2.4
![WIDE1DWGtransform_1 112 3](https://github.com/xtremestuff/protune-transforms/assets/76013032/54a440b9-baa0-4c5b-86c2-e7b443a9d634)

Very dark result, and since the mid gray is so high, I added 2.5 stops of linear gain.
![2 5 stop linear gain_1 112 4](https://github.com/xtremestuff/protune-transforms/assets/76013032/594c8995-cc14-4611-afe0-c7eb73f2b3d5)

Extremely pink results, and super saturated.

Desaturating seems to make it very useable, but it's still noticeably more pink than it's supposed to be.
![Wide_desat](https://github.com/xtremestuff/protune-transforms/assets/76013032/ae1f009e-e8e9-4de9-acf4-1b1c301e95bd)

Node tree is the following:
![Nodetree](https://github.com/xtremestuff/protune-transforms/assets/76013032/dbe327b8-a081-4dd5-8a47-b022f7a805bf)

**Node 1:** IDT (gopro native to DWG)
**Node 2:** 2.5 Stops of linear gain
**Node 3:** ODT (DWG/DWG to Rec709, Gamma2.4)

Care to chime in on what I'm doing wrong?

**Comment by RikshaDriver** [comment]
If you're using WIDE=1, this is equivalent to using the NATIVE values with auto white balancing. The camera is attempting to automatically perform white balancing and the tint/temperature are baked into the video.

If you're compressing the full gamut to a Rec.709 color space, you will need to perform some form of gamut mapping to bring the saturation down, otherwise everything will be oversaturated. What you're seeing is largely expected.

Also keep in mind that the transforms are using a standard LOGB value of 113, not 400, so the correction curves are not the same.

**Reply by markusfinholt** [reply]
Yeah I did use gamut compression to bring down the saturation, but I found there was no good sweetspot for an even desaturation of the most saturated areas.

Interesting that it has to autoWB. I would think it's more beneficial to have it apply whatever WB you set in the gopro settings. In 99% of cases, outdoor shots are shot with 5500k. So there is no way of having a set white-balance with sensor native color gamuts then? You have to use Native WB?

**Reply by markusfinholt** [reply]
Also, would just like to share that I did use it for a video, and got some really good results. So I'll likely keep using it if I find a quick workflow for working from Native. Thanks for the great work.

https://youtu.be/_V_M-xNajgA?si=lBvEdrxRLBZTapzy&t=222

**Reply by RikshaDriver** [reply]
Native has no white balancing. It's as captured directly by the sensor. You can use a white balance card at the start of each clip to get the white balancing right.

**Reply by markusfinholt** [reply]
Would it be possible to create a WIDE=1 LOGB=400 to DWG IDT? I don't have the math knowledge to do it.

**Reply by RikshaDriver** [reply]
I've created some LogB=400 transforms, however these are based on the Rec.709 color primaries.

I've done some further investigation and the Native matrix data is specific to the Hero 3 - Hero 5 Black range.

The other models use different sensors which require different matrices. I've updated the documentation accordingly.

Having said that, the Native transforms **may** work on the other series, but your mileage will vary and accuracy is not guaranteed.

I am working on support for the other series from Hero 6 - 12, however there is a degree of effort, testing and cost involved, which means that any future work on that front will be towards a paid plugin.

**Reply by markusfinholt** [reply]
I would happily pay a few bucks for a LOGB + WIDE to DWG transform for resolve.

**Reply by RikshaDriver** [reply]
Please see new transforms: https://xtremestuff.net/store/gp-tune-transform/

**Reply by markusfinholt** [reply]
Very happy with the GP-Tune transform so far. But I haven't found a good solution for gamut compression yet. The built in plugins in resolve are not compressing the gamut nearly enough, even with very strong compression, so I end up using the saturation knob to about 33 in Davinci Wide Gamut. I feel bad reccommending this workflow to noobs without a built in gamut compression setting in the plugin. Is this something you could develop?

**Reply by RikshaDriver** [reply]
You will find ACES has stronger gamut compression. I personally recommend the ACES workflow as the DaVinci Color Managed workflow has a lot of hidden gotchas and inconsistencies.

Also don't forget that sometimes the Labs settings don't stick. You may have a scenario where you are thinking that the footage was shot with WIDE=1 but it's actually still limited to Rec.709 gamut. If you use the WIDE or Native input transforms on this footage, the colors will be oversaturated. You can confirm that WIDE=1 is actually set by checking the metadata in your original clip:
https://gopro.github.io/labs/control/metadata/

Towards the end of the list, you should see something like: WIDE  1

![image](https://github.com/xtremestuff/protune-transforms/assets/5790757/cfd9a63f-e278-4a14-ae21-85fc218d9857)

**Reply by markusfinholt** [reply]
Interesting, didn't know that labs doesn't always stick to the recordings. I haven't had any issues with that yet I think.

What kind of gamut compression curve would you suggest I use for the wide/native shots? A strong gamut compression towards the more saturated parts, or a general desaturation method for the whole shot? The shot looks very evenly saturated, if that makes sense.

**Reply by RikshaDriver** [reply]
Using !M is permanent whereas oM doesn't withstand a reboot. There are also some quirks sometimes.

Generally ACES 1.3 with Reference Gamut Compress has worked well for me - sometimes a bit too well in the desaturation. If I need to desaturate a particular color (i.e. Blue), I use the Hue v Sat to bring down the saturation level for that color.

Circling back to your original post, have you checked that your original footage is in fact WIDE=1?

**Reply by markusfinholt** [reply]
My footage is Wide=1 yes, I tested it. I'll look into some more desaturation methods to see which one works. I'm not going to be leaving RCM anytime soon, as I've gotten too deep into the system to change to ACES.
